File: iosgeneratedcode/jcrystal/mobile/entities/enums/Color.swift
public enum Color : Int{
	case RED = 1000
	case BLUE = 2000
	case YELLOW = 3000
	var rawName: String?{
		switch self{
			case .RED : return "RED"
			case .BLUE : return "BLUE"
			case .YELLOW : return "YELLOW"
		}
	}
	var id: Int{
		return self.rawValue
	}
	public static func fromId(_ id : Int) -> Color{
		return Color(rawValue: id)!
	}
	static let values = [RED, BLUE, YELLOW]
}

File: src/main/java/dev/jcrystal/crystalfash/entities/Color.java
package dev.jcrystal.crystalfash.entities;

public enum Color {
	RED(1000),
	BLUE(2000),
	YELLOW(3000);
	public final long id;
	private Color(long id) {
		this.id = id;
	}
	public static Color fromId(long id) {
		for (Color val: Color.values()) {
			if (val.id==id) {
				return val;
			}
		}
		return null;
	}

}
